recognition/scoring.py

```diff
@@ -149,31 +156,61 @@ def run(self, output_to_report_dir=True):
+                    "Percent Word Accuracy": (self.out_percent_word_accuracy, None),
+                    "Ref. words": (None, self.out_ref_words),
+                    "Hyp. words": (None, self.out_hyp_words),
+                    "Aligned words": (None, self.out_aligned_words),
+                }
+
+                outputs_absolute: Dict[str, int] = {}
                 for line in f:
-                    s = line.split()
-                    if line.startswith("Percent Total Error"):
-                        self.out_wer.set(float(s[4][:-1]))
-                        self.out_num_errors.set(int("".join(s[5:])[1:-1]))
-                    elif line.startswith("Percent Correct"):
-                        self.out_percent_correct.set(float(s[3][:-1]))
-                        self.out_num_correct.set(int("".join(s[4:])[1:-1]))
-                    elif line.startswith("Percent Substitution"):
-                        self.out_percent_substitution.set(float(s[3][:-1]))
-                        self.out_num_substitution.set(int("".join(s[4:])[1:-1]))
-                    elif line.startswith("Percent Deletions"):
-                        self.out_percent_deletions.set(float(s[3][:-1]))
-                        self.out_num_deletions.set(int("".join(s[4:])[1:-1]))
-                    elif line.startswith("Percent Insertions"):
-                        self.out_percent_insertions.set(float(s[3][:-1]))
-                        self.out_num_insertions.set(int("".join(s[4:])[1:-1]))
-                    elif line.startswith("Percent Word Accuracy"):
-                        self.out_percent_word_accuracy.set(float(s[4][:-1]))
-                    elif line.startswith("Ref. words"):
-                        self.out_ref_words.set(int("".join(s[3:])[1:-1]))
-                    elif line.startswith("Hyp. words"):
-                        self.out_hyp_words.set(int("".join(s[3:])[1:-1]))
-                    elif line.startswith("Aligned words"):
-                        self.out_aligned_words.set(int("".join(s[3:])[1:-1]))
+                    key: Optional[str] = ([key for key in output_variables if line.startswith(key)] or [None])[0]
+                    if not key:
+                        continue
+                    pattern = rf"^{re.escape(key)}\s*=\s*((\S+)%)?\s*(\(\s*(\d+)\))?$"
+                    m = re.match(pattern, line)
+                    assert m, f"Could not parse line: {line!r}, does not match to pattern r'{pattern}'"
+                    absolute_s = m.group(4)
+                    if not absolute_s:
+                        assert not output_variables[key][1], f"Expected absolute value for {key}"
+                        continue
+                    outputs_absolute[key] = int(absolute_s)
+
+                num_ref_words = outputs_absolute["Ref. words"]
+                assert "Percent Total Error" in outputs_absolute, "Expected absolute numbers for Percent Total Error"
```

Comment:
This assertion is done for `"Percent Total Error"`, but not `"Ref. words"` above. Is there a reason to treat them differently?

Reply:
Not really, I can add that.